enumerable.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each(arr)
+    x = 0
+    while x < arr.length
+      yield(arr[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index(arr)
+    i = 0
+    while i < arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+  end
+
+  def my_select(arr)
+    y = 0
+    new_arr = []
+    while y < arr.length
+      new_arr.push(arr[y]) if yield(arr[y])
+      y += 1
+    end
+    new_arr
+  end
+
+  def my_all?(arr)
+    w = 0
+    while w < arr.length
+      return false unless yield(arr[w])
+
+      w += 1
+    end
+    true
```

Comment:
- [ ]  Enumerables `my_all?` returns true if the block never returns false or nil
`[1,2,3,4,5].my_all?{ |num| num < 10 }` gives `error` instead of `true`

- [ ]  Enumerables `my_all?` when no block or argument is given returns true when none of the collection members are false or nil
`[1,2,3,4,5].my_all?` should return `true`

- [ ] Enumerables `my_all?` when a class is passed as an argument returns true if all of the collection is a member of such class
`[1,2,3,4,5].my_all?(Integer)` should return `true`

- [ ] Enumerables `my_all?` when a Regex is passed as an argument returns true if all of the collection matches the Regex
`['saheed', 'oladele', 'suretrust'].my_all?(/d/)` should return `true`

- [ ] Enumerables `my_all?` when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
`[1,2,3,4,5].my_all?(3)` should return `false`